src/backend/metal/src/command.rs

```diff
@@ -461,11 +467,22 @@ impl RawCommandBuffer<Backend> for CommandBuffer {
 
     fn update_buffer(
         &mut self,
-        _buffer: &native::Buffer,
-        _offset: u64,
-        _data: &[u8],
+        dst: &native::Buffer,
+        offset: u64,
+        data: &[u8],
     ) {
-        unimplemented!()
+        let src = self.device().new_buffer_with_data(data.as_ptr() as _, data.len() as _, metal::MTLResourceOptions::StorageModePrivate);
```

Comment:
when is this buffer freed?

Reply:
When the command buffer drops reference to it.

Reply:
where is it keeping that reference?